### PR TITLE
KAFKA-18077: Remove deprecated JmxReporter(String)

### PR DESCRIPTION
**Pull request by clarkwtc**
Remove the following code about the deprecated JmxReporter(String).
> - Remove initial JmxReporter constructor with string and refactor default constructor.
> - Remove related test fucntion for testDeprecatedJmxPrefixWithDefaultMetrics.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by chia7712 — COMMENTED** [review]
@clarkwtc please update `upgrade.html` also

**Comment by clarkwtc** [comment]
@chia7712 I have updated the `upgrade.html`, please review it.

**Review by chia7712 — APPROVED** [review]

File: clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java [hunk 1]
```diff
@@ -69,20 +69,7 @@ public class JmxReporter implements MetricsReporter {
...
-     * Create a JMX reporter that prefixes all metrics with the given string.
-     *  @deprecated Since 2.6.0. Use {@link JmxReporter#JmxReporter()}
-     *  Initialize JmxReporter with {@link JmxReporter#contextChange(MetricsContext)}
-     *  Populate prefix by adding _namespace/prefix key value pair to {@link MetricsContext}
-     */
-    @Deprecated
-    public JmxReporter(String prefix) {
-        this.prefix = prefix != null ? prefix : "";
-    }
+    public JmxReporter() {}
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java]: Please initialize "prefix" to empty string and then remove this default constructor as well.

**Review by chia7712 — COMMENTED** [review]
@clarkwtc thanks for your patch. please rebase code

File: clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java (hunk 1 above)
Comment (clarkwtc) [reply clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java]: @chia7712 I got it; I missed initializing "prefix" to empty string, thanks.

**Pull request merged**